### DISCUSSION TITLE
Object type composition

### DISCUSSION THREAD

**Discussion by aganm** [mapeditor/tiled#2950] [Ideas]
Is there plans to add object type composition?

In the same manner that I can combine the Flip values
![image](https://user-images.githubusercontent.com/20380758/101564803-ce5e5400-3999-11eb-89d4-3d625981313d.png)

I would like to be able to combine object types
![image](https://user-images.githubusercontent.com/20380758/101564828-dd450680-3999-11eb-8002-f71f06263900.png)

What this would do is combine the custom properties of each type into a single object such as
![image](https://user-images.githubusercontent.com/20380758/101565109-63f9e380-399a-11eb-9204-3c5c31b0e917.png)

**Comment by bjorn** [comment]
The closest planned feature would be nested/compound custom properties (#489), currently scheduled for Tiled 1.6, so it's one of the things I want to work on once 1.5 is done.

So, instead of setting multiple types on the object, you could define a Sound and a Weapon custom compound type and then add properties of those types to your object (or its type).

**Reply by bjorn** [reply]
Just noting that [custom classes](https://doc.mapeditor.org/en/stable/manual/custom-properties/#custom-types), added in Tiled 1.8, now provide a way of doing almost exactly this. So not via the "Type" property (renamed to "Class" in 1.9), but by adding properties based on custom classes.
